core/src/main/java/com/linecorp/armeria/client/WebClient.java

```diff
@@ -298,6 +299,25 @@ default WebClientRequestPreparation prepare() {
         return new WebClientRequestPreparation(this);
     }
 
+    /**
+     * Appends query params to the current path and returns the modified path.
+     */
+    default String addQueryParams(String path, QueryParams params) {
```

Comment:
I'm not sure whether we need to expose this method as a public API.
How about hiding this by moving to a package-private class?
```
package com.linecorp.armeria.client;

final class WebClientUtil {

    static String addQueryParams(String path, QueryParams params) {
    ...
}
```

Reply:
Done.